c/planarity.c

```diff
@@ -19,12 +19,156 @@ void ProjectTitle()
-    		"\nContact info: jboyer at acm.org"
-            "\n=================================================="
-            "\n");
+	return "\n=================================================="
+			"\nThe Edge Addition Planarity Suite version 3.0.2.0"
+			"\nCopyright (c) 1997-2024 by John M. Boyer"
+			"\nContact info: jboyer at acm.org"
+			"\n=================================================="
+			"\n";
+}
+
+/****************************************************************************
+ ALGORITHM FLAGS/SPECIFIERS
+****************************************************************************/
+
+char * GetAlgorithmFlags(void)
+{
+	return  "C = command (algorithm implementation to run)\n"
+			"    -p = Planar embedding and Kuratowski subgraph isolation\n"
+			"    -d = Planar graph drawing by visibility representation\n"
+			"    -o = Outerplanar embedding and obstruction isolation\n"
+			"    -2 = Search for subgraph homeomorphic to K_{2,3}\n"
+			"    -3 = Search for subgraph homeomorphic to K_{3,3}\n"
+			"    -4 = Search for subgraph homeomorphic to K_4\n"
+			"\n";
+}
+
+char * GetAlgorithmSpecifiers(void)
+{
+	return  "P. Planar embedding and Kuratowski subgraph isolation\n"
+			"D. Planar graph drawing by visibility representation\n"
+			"O. Outerplanar embedding and obstruction isolation\n"
+			"2. Search for subgraph homeomorphic to K_{2,3}\n"
+			"3. Search for subgraph homeomorphic to K_{3,3}\n"
+			"4. Search for subgraph homeomorphic to K_4\n";
+}
+
+char * GetAlgorithmChoices(void)
+{
+	return "pdo234";
+}
+
+char * GetSupportedOutputFormats(void)
+{
+	return "gam";
+}
+
+/****************************************************************************
+ helpMessage()
+ ****************************************************************************/
+
+int helpMessage(char *param)
+{
+	Message(GetProjectTitle());
+
+	if (param == NULL)
+	{
+		Message(
+			"'planarity': if no command-line, then menu-driven\n"
+			"'planarity (-h|-help)': this message\n"
+			"'planarity (-h|-help) -menu': more help with menu-based command line\n"
+			"'planarity (-i|-info): copyright and license information\n"
+			"'planarity -test [-q] [samples dir]': runs tests (optional quiet mode)\n"
+			"\n"
+		);
+
+		Message(
+			"Common usages\n"
+			"-------------\n"
+			"planarity -s -q -p infile.txt embedding.out [obstruction.out]\n"
+			"Process infile.txt in quiet mode (-q), putting planar embedding in \n"
+			"embedding.out or (optionally) a Kuratowski subgraph in Obstruction.out\n"
+			"Process returns 0=planar, 1=nonplanar, -1=error\n"
+			"\n"
+			"planarity -s -q -d infile.txt embedding.out [drawing.out]\n"
+			"If graph in infile.txt is planar, then put embedding in embedding.out \n"
+			"and (optionally) an ASCII art drawing in drawing.out\n"
+			"Process returns 0=planar, 1=nonplanar, -1=error\n"
+		);
+	}
+
+	else if (strcmp(param, "-i") == 0 || strcmp(param, "-info") == 0)
+	{
+		Message(
+			"The Edge Addition Planarity Suite version 3.0.2.0\n"

```

Comment:
Now that you've cleaned up putting the project title into a string, you can invoke it here to clean up my prior bad act of duplicating the message.